crates/astria-core/src/primitive/v1/asset/denom.rs

```diff
@@ -75,6 +75,24 @@ impl Denom {
         };
         trace
     }
+
+    /// Calculates the length of the display formatted [Denom] without allocating a String.
+    #[must_use]
+    pub fn display_len(&self) -> Option<usize> {
```

Comment:
In general we use checked addition, but in this case returning `usize` is more than acceptable.

On any currently available machine you will not blow through 2^64-1 bytes of RAM.